api/views/oauth.py

```diff
@@ -47,10 +47,32 @@ def post(self, request):
         post_data = FormOrJsonParser().parse_body(request)
         # Grab the application and other details again
         redirect_uri = post_data["redirect_uri"]
-        scope = post_data["scope"]
-        application = Application.objects.get(client_id=post_data["client_id"])
+        scope = post_data["scope"] or "read"
+        try:
+            application = Application.objects.get(client_id=post_data["client_id"])
+        except Application.DoesNotExist:
+            return JsonResponse(
+                {
+                    "error": "invalid_client",
+                    "error_description": "Client authentication failed due to unknown client, no client authentication included, or unsupported authentication method.",
+                },
+                status=401,
+            )
+
+        if not application.is_scope_subset(scope):
```

Comment:
So when I used Tusky, it didn't set the application scope at all and just relied on the authentication scope, from what I can see. I don't see any security advantage from respecting the application scope here - let's just go with whatever they request.

The scope subset code should probably move to Token so we can use it for view auth checking.